api/opentrons/server/endpoints/update.py

```
@@ -42,3 +44,81 @@
     robot._driver._setup()
 
     return res
+
+
+async def update_module_firmware(request):
+    """
+     This handler accepts a POST request with Content-Type: multipart/form-data
+     and a file field in the body named "module_firmware". The file should
+     be a valid HEX image to be flashed to the atmega32u4. The received file is
+     sent via USB to the board and flashed by the avr109 bootloader. The file
+     is then deleted and a success code is returned
+    """
+    log.debug('Update Firmware request received')
+    data = await request.post()
+    module_serial = request.match_info['serial']
```

If the input to this endpoint is invalid, it should return a 400 error. Here, if the specified serial doesn't exist on the robot it will return a 200 OK with an error message. We should have a way to surface the module not being present as an error.

Also, we should probably check the file or at least the results of the upload and do something similar. If avrdude fails because the file is bad, that should be a 400 error; for any other reason, probably a 500.